libcodechecker/analyze/log_parser.py

```diff
@@ -210,18 +220,121 @@ def get_compiler_target(parseLogOptions, compiler):
         err = get_compiler_err(cmd)
     else:
         err = load_compiler_info(parseLogOptions.compiler_target_file,
-                                 compiler)
+                                 compiler,
+                                 'target')
 
     if parseLogOptions.output_path is not None:
-        dump_compiler_info(parseLogOptions.output_path,
-                           compiler_target_dump_file,
-                           {compiler: err})
+        dump_compiler_info(compiler_info_dump_file,
+                           compiler,
+                           'target',
+                           err)
     return parse_compiler_target(err)
 
 
-def remove_file_if_exists(filename):
-    if os.path.isfile(filename):
-        os.remove(filename)
+def get_compiler_standard(parseLogOptions, compiler, lang):
+    """
+    Returns the default compiler standard of the given compiler. The standard
+    is determined by the values of __STDC_VERSION__ and __cplusplus predefined
+    macros. These values are integers indicating the date of the standard.
+    However, GCC supports a GNU extension for each standard. For sake of
+    generality we return the GNU extended standard, since it should be a
+    superset of the non-extended one, thus applicable in a more general manner.
+    """
+    VERSION_C = u"""
+#include <stdio.h>
+
+int main()
+{
+  int version = 90;
+
+#if defined(__STDC_VERSION__)
+#  if __STDC_VERSION__ >= 201710L
+  version = 17;
+#  elif __STDC_VERSION__ >= 201112L
+  version = 11;
+#  elif __STDC_VERSION__ >= 199901L
+  version = 99;
+#  elif __STDC_VERSION__ >= 199409L
+  version = 95;
+#  endif
+#endif
+
+  printf("%d\\n", version);
+
+  return 0;
+}
+    """
+
+    VERSION_CPP = u"""
+#include <iostream>
+
+int main()
+{
+  int version = 98;
+
+#if defined(__cplusplus)
+#  if __cplusplus >= 201703L
+  version = 17;
+#  elif __cplusplus >= 201402L
+  version = 14;
+#  elif __cplusplus >= 201103L
+  version = 11;
+#  elif __cplusplus >= 199711L
+  version = 98;
+#  endif
+#endif
+
+  std::cout << version << std::endl;
+
+  return 0;
+}
+    """
+
+    standard = ""
+    if parseLogOptions.compiler_info_file is None:
+        _, temp_file = tempfile.mkstemp()
+        test_file = temp_file + ('.c' if lang == 'c' else '.cpp')
+        os.rename(temp_file, test_file)
```

Comment:
I think removing a temp file should be done in the following form:
```py
_, temp_file = tempfile.mkstemp()
try:
    ...
finally:
    remove_file_if_exists(temp_file)
    remove_file_if_exists(test_file)
```
What do you think @gyorb?

Reply:
Using `NamedTemporaryFile` in a with statement with a suffix (no renaming is needed) would be easier and it would remove the file when it is out of context.